node/src/actors/chain_manager/mod.rs

```diff
@@ -278,54 +271,55 @@ impl ChainManager {
-                    .chain_state
-                    .data_request_pool
-                    .update_data_request_stages();
-                for reveal in reveals {
-                    // Send AddTransaction message to self
-                    // And broadcast it to all of peers
-                    ctx.address().do_send(AddTransaction {
-                        transaction: reveal,
-                    })
-                }
-                // Persist finished data requests into storage
-                let to_be_stored = self.chain_state.data_request_pool.finished_data_requests();
-                to_be_stored.into_iter().for_each(|dr| {
-                    self.persist_data_request(ctx, &dr);
-                    if info_flag {
+                if let StateMachine::Synced = self.sm_state {
+                    // Persist finished data requests into storage
+                    let to_be_stored = self.chain_state.data_request_pool.finished_data_requests();
+                    to_be_stored.into_iter().for_each(|dr| {
+                        self.persist_data_request(ctx, &dr);
                         show_info_tally(&self.chain_state.unspent_outputs_pool, dr, block_epoch);
-                    }
-                });
+                    });
 
-                if info_flag {
                     show_info_dr(&self.chain_state.data_request_pool, &block);
 
-                    debug!("{:?}", block);
+                    log::trace!("{:?}", block);
                     debug!("Mint transaction hash: {:?}", block.txns[0].hash());
-                }
 
-                // Insert candidate block into `block_chain` and persist it
-                self.chain_state.block_chain.insert(block_epoch, block_hash);
-                self.persist_item(ctx, InventoryItem::Block(block.clone()));
+                    let reveals = self
+                        .chain_state
+                        .data_request_pool
+                        .update_data_request_stages();
+
+                    for reveal in reveals {
+                        // Send AddTransaction message to self
+                        // And broadcast it to all of peers
+                        ctx.address().do_send(AddTransaction {
+                            transaction: reveal,
+                        })
+                    }
 
-                // Persist chain_info into storage
-                if let StateMachine::Synced = self.sm_state {
+                    self.persist_item(ctx, InventoryItem::Block(block.clone()));
+
+                    // Persist chain_info into storage
                     self.persist_chain_state(ctx);
-                }
 
-                // Send notification to JsonRpcServer
-                JsonRpcServer::from_registry().do_send(NewBlock { block })
+                    // Send notification to JsonRpcServer
+                    JsonRpcServer::from_registry().do_send(NewBlock {
+                        block: block.clone(),
```

Comment:
Unfortunately, it is. Currently the line of thought is: *consolidate_block* shouldn't take ownership of the block because when validating and consolidating block batches, we need to keep ownership of those blocks in order to persist them. The ideal would be to move this *notification-related/actor-coupled* code out of the consolidation function.